### PR TITLE
Feature/scat 2303 upload document

### PR DESCRIPTION
**Pull request by adrianmilne**
### JIRA link (if applicable) ###
SCAT-2303

### Change description ###

Adds document upload/retrieval endpoints - as outlined in open PR here - https://github.com/Crown-Commercial-Service/ccs-scale-api-definitions/pull/98

Note: this is referenced in pom.xml and needs updating once API changes are merged

### Does this PR introduce a breaking change?

- [ ] Yes
- [ X] No

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java [hunk 1]
```diff
@@ -91,4 +96,50 @@ public String deleteSupplier(@PathVariable("procID") final Integer procId,
...
+    log.info("getDocumentSummaries invoked on behalf of principal: {}", principal);
+
+    return procurementEventService.getDocumentSummaries(procId, eventId);
+  }
+
+  @PutMapping("/{eventID}/documents")
+  public DocumentSummary uploadDocument(@PathVariable("procID") final Integer procId,
+      @PathVariable("eventID") final String eventId,
+      @RequestParam("data") final MultipartFile multipartFile,
+      @RequestParam("audience") final String audience,
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java:115]: The generated enum (`DocumentAudienceType`) has a `fromValue` method annotated with `@JsonCreator` - so I think you can go straight to that type

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java [hunk 2]
```diff
@@ -91,4 +96,50 @@ public String deleteSupplier(@PathVariable("procID") final Integer procId,
...
+      final JwtAuthenticationToken authentication) {
+
+    var principal = getPrincipalFromJwt(authentication);
+    log.info("getDocumentSummaries invoked on behalf of principal: {}", principal);
+
+    return procurementEventService.getDocumentSummaries(procId, eventId);
+  }
+
+  @PutMapping("/{eventID}/documents")
+  public DocumentSummary uploadDocument(@PathVariable("procID") final Integer procId,
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java:112]: Can we use `@Valid` / `@Validated` (at the request param level - for audience mainly I think - I get a 500 if I omit it in the request)

**Review by tommyb82 — COMMENTED** [review]
Getting a 500 trying to upload a document - probably just user error:

```

java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "uk.gov.crowncommercial.dts.scale.cat.model.jaggaer.SellerAttachmentsList.getAttachment()" is null
	at uk.gov.crowncommercial.dts.scale.cat.service.ProcurementEventService.getDocumentSummaries(ProcurementEventService.java:390) ~[classes/:na]
	at uk.gov.crowncommercial.dts.scale.cat.service.ProcurementEventService.uploadDocument(ProcurementEventService.java:429) ~[classes/:na]
	at uk.gov.crowncommercial.dts.scale.cat.controller.EventsController.uploadDocument(EventsController.java:124) ~[classes/:na]
	at uk.gov.crowncommercial.dts.scale.cat.controller.EventsController$$FastClassBySpringCGLIB$$d660a1b.invoke(<generated>) ~[classes/:na]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.13.jar:5.3.13]
```

**Review by tommyb82 — APPROVED** [review]
Superb stuff - tested all 3 endpoints with mutliple documents - all working great now 🥇

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java (hunk 1 above)
Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java:115]: I had that originally - but it only worked if the param was in upper case - e.g. `BUYER` (even though it has lower case constructors generated from the API) - I'll give it another try though

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java (hunk 2 above)
Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java:112]: @tommyb82 - tried adding that `@RequestParam("audience") @Valid final String audience` - but didn't seem to make any difference - still get a 500 `"Required request parameter 'audience' for method parameter type String is not present"` message

**Pull request merged**